### PR TITLE
Splitting the monolith recomendation micro

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: api_gateway/views/recomendation.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+bp = Blueprint('recomendation', __name__, url_prefix='/recomendations')
+
+RECOMENDATOR_HOST = os.getenv("RECOMENDATOR_HOST", "localhost")
+RECOMENDATOR_PORT = os.getenv("RECOMENDATOR_PORT", "50051")
+
+GRPC_CHANNEL = grpc.insecure_channel(f"{RECOMENDATOR_HOST}:{RECOMENDATOR_PORT}")
+GRPC_STUB = recomendations_pb2_grpc.RecomendationsControllerStub(GRPC_CHANNEL)
+
+@bp.get('/<int:user_id>')
```
Comment (genismoreno) [inline api_gateway/views/recomendation.py:19]: Let's change this to borrower or borrower_id to be consistent with the model

File: recomendator/recomendations/admin.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
+
+# Register your models here.
```
Comment (genismoreno) [inline recomendator/recomendations/admin.py:3]: Is this necessary?

File: recomendator/recomendator/settings.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+USE_L10N = True
+
+USE_TZ = True
+
+
+# Static files (CSS, JavaScript, Images)
+# https://docs.djangoproject.com/en/2.2/howto/static-files/
+
+STATIC_URL = '/static/'
```
Comment (genismoreno) [inline recomendator/recomendator/settings.py:137]: is static configured?

File: recomendator/recomendations/models.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4,
+                          help_text="Unique ID for this particular book across whole library")
+    book = models.ForeignKey('Book', on_delete=models.RESTRICT, null=True)
+    imprint = models.CharField(max_length=200)
+    due_back = models.DateField(null=True, blank=True)
+    borrower = models.ForeignKey(LibraryUser, on_delete=models.SET_NULL, null=True, blank=True)
+
+    @property
+    def pk(self):
```
Comment (genismoreno) [inline recomendator/recomendations/models.py]: Isn't django doing this automatically? At least in the catalog it's working

**Review by genismoreno — CHANGES_REQUESTED** [review]

File: api_gateway/views/recomendation.py (hunk 1 above)
Comment (ghost) [reply api_gateway/views/recomendation.py:19]: But we are not checking by borrower for BookInstances but for user in the system :/

File: recomendator/recomendations/models.py (hunk 4 above)
Comment (ghost) [reply recomendator/recomendations/models.py]: If you mean the pk property, it does, but CQRS will fail by this as the id is an uuid and it does'nt know how to serialize it and neither can be fixed in the custom serializer

File: recomendator/recomendations/admin.py (hunk 2 above)
Comment (ghost) [reply recomendator/recomendations/admin.py:3]: I'll add a model to the service as 'Recomendations' to persist the reomendations made, the admin could be used to check and modifiy them, but I don't have huge pros with this.
At the end is the auto generated code by django

File: api_gateway/views/recomendation.py (hunk 1 above)
Comment (genismoreno) [reply api_gateway/views/recomendation.py:19]: Sure, the model has a foreign key to the user model, but it's called borrower. I just changed it my PR. I don't say it's wrong, but just to be consistent everywhere

**Review by aydevosotros — APPROVED** [review]

File: recomendator/recomendator/settings.py (hunk 3 above)
Comment (aydevosotros) [reply recomendator/recomendator/settings.py:137]: looks like a django default

Comment (ghost) [reply recomendator/recomendator/settings.py:137]: It is

File: api_gateway/views/recomendation.py (hunk 1 above)
Comment (ghost) [reply api_gateway/views/recomendation.py:19]: I dont agree, its true that BookInstance FK is borrower, but in this endpoints we are calling for recomendations to a given user, not for a borrower. It is possible to recommend books to users that have not borrow a single book and then they have no entry in  BookInstance

File: api_gateway/views/author.py [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import os
+import json
```
Comment (genismoreno) [inline api_gateway/views/author.py]: I think it's unused

File: api_gateway/views/loan.py [hunk 6]
```diff
@@ -1,5 +1,6 @@
 import os
 from uuid import UUID
+import json
```
Comment (genismoreno) [inline api_gateway/views/loan.py]: Same here

**Review by genismoreno — APPROVED** [review]

**Pull request merged**